### PR TITLE
io.github.fsobolev.TimeSwitch

### PR DESCRIPTION
**Pull request by fsobolev**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: io.github.fsobolev.TimeSwitch.json [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        "*.a"
+    ],
+    "modules" : [
+    	{
+            "name": "gsound",
+            "buildsystem": "meson",
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://gitlab.gnome.org/GNOME/gsound.git"
```
Comment (hfiguiere) [inline io.github.fsobolev.TimeSwitch.json:34]: you need a specific revision here.

File: io.github.fsobolev.TimeSwitch.json [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                }
+            ],
+            "modules": [
+                {
+                    "name": "libcanberra",
```
Comment (hfiguiere) [inline io.github.fsobolev.TimeSwitch.json:39]: you should use the one in shared-modules. it contains necessary patches.

File: io.github.fsobolev.TimeSwitch.json [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "http://people.freedesktop.org/~mccann/dist/sound-theme-freedesktop-0.8.tar.bz2",
+                    "sha256": "cb518b20eef05ec2e82dda1fa89a292c1760dc023aba91b8aa69bafac85e8a14"
+                }
+            ],
+            "modules": [
+                {
+                    "name": "intltool",
```
Comment (hfiguiere) [inline io.github.fsobolev.TimeSwitch.json:66]: you should use the on in shared-modules

File: io.github.fsobolev.TimeSwitch.json [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    "runtime-version" : "43",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "timeswitch",
+    "finish-args" : [
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--device=dri",
+        "--socket=wayland",
+        "--talk-name=org.gnome.SessionManager",
+        "--system-talk-name=org.freedesktop.login1",
```
Comment (hfiguiere) [inline io.github.fsobolev.TimeSwitch.json:13]: you didn't explan why you need thes dbus permissions.

Comment (fsobolev) [reply io.github.fsobolev.TimeSwitch.json:13]: The app allows to power off, reboot and suspend system. For that it uses org.gnome.SessionManager when running in GNOME, and org.freedesktop.login1 when running in another DE.

**Comment by barthalion** [comment]
bot, build io.github.fsobolev.TimeSwitch

**Comment by fsobolev** [comment]
Sorry if it's not the right place to ask, I accepted the invitation, but I still don't have write access to the master branch of created repository. Did I do something wrong?

**Comment by hfiguiere** [comment]
it's protected, you can't push directly to. Use PR and you can merge them if it passes CI.

**Pull request closed without merging**